internal/config/config_file.go

```diff
@@ -39,40 +40,99 @@ func ConfigDir() string {
 
 	// If the path does not exist try migrating config from default paths
 	if _, err := os.Stat(path); errors.Is(err, os.ErrNotExist) {
-		autoMigrateConfigDir(path)
+		_ = autoMigrateConfigDir(path)
 	}
 
 	return path
 }
 
+func StateDir() string {
+	if path := os.Getenv(XDG_STATE_HOME); path != "" {
+		path = filepath.Join(path, "gh")
+		if !dirExists(path) {
+			_ = os.MkdirAll(path, 0755)
```

Comment:
Is this mkdir necessary since `migrateFile` does it internally? It feels weird that just reading `StateDir` creates a filesystem entry even if no migration is happening